src/Internal/IntOperations.elm

```diff
@@ -0,0 +1,236 @@
+module Internal.IntOperations exposing (IntOperations, int32Operations, int64Operations)
+
+import Bitwise
+import Protobuf.Types.Int64 as Int64 exposing (Int64)
+
+
+type alias IntOperations int =
+    { zigZag : int -> int
+    , zagZig : int -> int
+    , toUnsigned : int -> int
+    , fromUnsigned : int -> int
+    , split7Bit : int -> ( Int, int )
+    , add7Bit : Int -> int -> int
+    , zero : int
+    }
+
```

Comment:
I think it should look like this:

```
type alias IntOperations int =
    { toZigZag : int -> int
    , fromZigZag : int -> int
    , toUnsigned : int -> int
    , fromUnsigned : int -> int
    , popByte : int -> ( Int, int )
    , pushByte : Int -> int -> int
    , fromByte : Int -> int
    }
```

The logic of getting the base128 from `popByte` can be in the encoder itself
`fromByte` can be used to construct `0` or replace `config.add7Bit octet config.zero`

Reply:
Ah wait, let me get back to this - think it missing a point with the base128

Reply:
So probably this makes more sense:
```
type alias IntOperations int =
    { toZigZag : int -> int
    , fromZigZag : int -> int
    , toUnsigned : int -> int
    , fromUnsigned : int -> int
    , popBase64 : int -> ( Int, int )
    , pushBase64 : Int -> int -> int
    , fromBase64 : Int -> int
    }
```

If it helps you can define a shared `toBase128 : Int -> Int` (doing the `Bitwise.and`

Reply:
Good idea, I like the interface better than before. Done :)